fenix/app/src/main/java/org/mozilla/fenix/components/appstate/AppStoreReducer.kt

```diff
@@ -11,11 +11,11 @@ import mozilla.components.service.pocket.ext.recordNewImpression
 import org.mozilla.fenix.components.AppStore
 import org.mozilla.fenix.ext.filterOutTab
 import org.mozilla.fenix.ext.getFilteredStories
-import org.mozilla.fenix.gleanplumb.state.MessagingReducer
 import org.mozilla.fenix.home.pocket.PocketRecommendedStoriesSelectedCategory
 import org.mozilla.fenix.home.recentsyncedtabs.RecentSyncedTabState
 import org.mozilla.fenix.home.recentvisits.RecentlyVisitedItem
 import org.mozilla.fenix.home.recentvisits.RecentlyVisitedItem.RecentHistoryGroup
+import org.mozilla.fenix.messaging.state.MessagingReducer
```

Comment:
I was wondering if we should also uplift, the state bits to AC, like `MessagingMiddleware`and `MessagingState` as this will need to be duplicated in Focus when we need to used it.

Reply:
No suggesting that it should be added as part of this PR as it's already big, but maybe something that we could consider for re-usability 😄 .

Reply:
Do you think that those suggestions can fall under the scope of [this bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1823736) ? 
If so, I can add it in its description

Reply:
Sorry, for the delay I think we may need another bug, as I was referring to uplift the state part of the messages, like `MessagingMiddleware`, and `MessagingState` which are app agnostic.